src/main/java/io/supertokens/dashboard/Dashboard.java

```diff
@@ -0,0 +1,244 @@
+/*
+ *    Copyright (c) 2023, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ *    This software is licensed under the Apache License, Version 2.0 (the
+ *    "License") as published by the Apache Software Foundation.
+ *
+ *    You may not use this file except in compliance with the License. You may
+ *    obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+
+package io.supertokens.dashboard;
+
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.regex.Pattern;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import io.supertokens.Main;
+import io.supertokens.dashboard.exceptions.UserSuspendedException;
+import io.supertokens.emailpassword.PasswordHashing;
+import io.supertokens.featureflag.EE_FEATURES;
+import io.supertokens.featureflag.FeatureFlag;
+import io.supertokens.featureflag.exceptions.FeatureNotEnabledException;
+import io.supertokens.pluginInterface.dashboard.DashboardSessionInfo;
+import io.supertokens.pluginInterface.dashboard.DashboardUser;
+import io.supertokens.pluginInterface.dashboard.exceptions.DuplicateEmailException;
+import io.supertokens.pluginInterface.dashboard.exceptions.DuplicateUserIdException;
+import io.supertokens.pluginInterface.dashboard.exceptions.UserIdNotFoundException;
+import io.supertokens.pluginInterface.dashboard.sqlStorage.DashboardSQLStorage;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.exceptions.StorageTransactionLogicException;
+import io.supertokens.storageLayer.StorageLayer;
+import io.supertokens.utils.Utils;
+import jakarta.annotation.Nullable;
+
+public class Dashboard {
+    public static final int MAX_NUMBER_OF_FREE_DASHBOARD_USERS = 1;
+    public static final long DASHBOARD_SESSION_DURATION = 2592000000L; // 30 days in milliseconds
+
+    public static void signUpDashboardUser(Main main, String email, String password)
+            throws StorageQueryException, DuplicateEmailException, FeatureNotEnabledException {
+
```

Comment:
first check if the user already exists and if it does, throw DuplicateEmailException